pandas/core/array_algos/npcompat.py

```diff
@@ -0,0 +1,34 @@
+"""
+Implementations of high-level numpy functions that are ExtensionArray-compatible.
+"""
+import numpy as np
+
+from pandas._typing import ArrayLike
+
+
+def tile(arr: ArrayLike, shape) -> ArrayLike:
+    raise NotImplementedError
+
+
+def broadcast_to(arr: ArrayLike, shape, orient=None) -> ArrayLike:
+    if isinstance(arr, np.ndarray):
+        values = arr
+    else:
+        # ExtensionArray
+        values = arr._values_for_factorize()[0]
+
+    # TODO: if we are ndim==size==1 it shouldnt matter whether rowlike/columnlike?
+    if values.ndim == 1 and orient is not None:
+        # SUpport treating a 1-dimensional array as either a row or column
+        assert orient in ["rowlike", "columnlike"]
+        if orient == "rowlike":
+            values = values.reshape(1, -1)
+        else:
+            values = values.reshape(-1, 1)
+
+    btvalues = np.broadcast_to(values, shape)
```

Comment:
Is the result `btvalues` being returned to a user? `broadcast_to` is typically (always?) a readonly view, and I don't think we want to be returning readonly results.

Reply:
the way its used in this PR, it is not returned to a user, but used as an intermediate object in the arithmetic op